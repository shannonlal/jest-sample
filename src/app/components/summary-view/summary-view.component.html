<div>
    <p *ngFor="let phone of phones">
        {{phone}}
    </p>
</div>
