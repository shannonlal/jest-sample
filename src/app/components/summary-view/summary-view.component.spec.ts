import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { SummaryViewComponent } from './summary-view.component';

describe('SummaryViewComponent', () => {
  let component: SummaryViewComponent;
  let fixture: ComponentFixture<SummaryViewComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ SummaryViewComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(SummaryViewComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create summary-view component', () => {
    expect(component).toBeTruthy();
  });

  it('should render snapshot component with one name', () => {

    component.phones = ['TestPhone'];
    fixture.detectChanges();

    expect( fixture ).toMatchSnapshot();
  });

  it('should render snapshot component with 3 phones', () => {

    component.phones = ['TestPhone', 'Test Phone B', 'Test Phone C'];
    fixture.detectChanges();

    expect( fixture ).toMatchSnapshot();
  });

  it('should render component with one phone', () => {

    component.phones = ['TestPhone'];
    fixture.detectChanges();

    const element: HTMLElement = fixture.nativeElement;
    const pTags = element.querySelectorAll('p');
    expect( pTags.length ).toBe(1);
  });

  it('should render component with one phone', () => {

    component.phones = ['TestPhone', 'Phone B', 'Phone C'];
    fixture.detectChanges();

    const element: HTMLElement = fixture.nativeElement;
    const pTags = element.querySelectorAll('p');
    expect( pTags.length ).toBe(3);
  });

});
