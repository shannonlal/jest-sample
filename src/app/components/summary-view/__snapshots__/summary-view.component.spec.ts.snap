// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryViewComponent should render snapshot component with 3 phones 1`] = `
<app-summary-view
  phones={[Function Array]}
>
  <div>
    
    <p>
       TestPhone 
    </p>
    <p>
       Test Phone B 
    </p>
    <p>
       Test Phone C 
    </p>
  </div>
</app-summary-view>
`;

exports[`SummaryViewComponent should render snapshot component with one name 1`] = `
<app-summary-view
  phones={[Function Array]}
>
  <div>
    
    <p>
       TestPhone 
    </p>
  </div>
</app-summary-view>
`;
